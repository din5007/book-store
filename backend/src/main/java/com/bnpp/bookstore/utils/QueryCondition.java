package com.bnpp.bookstore.utils;

import io.micrometer.common.util.StringUtils;
import org.springframework.stereotype.Component;

@Component("queryUtil")
public class QueryCondition {

  public boolean checkNull(String field) {
    return StringUtils.isBlank(field);
  }
}
