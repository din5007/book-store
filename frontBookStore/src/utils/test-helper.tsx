import { StoreProvider } from "easy-peasy";
import { render } from '@testing-library/react';
import { BrowserRouter } from "react-router-dom";


function customRender(component : any , store?: any) {

    return render(<StoreProvider store={store}>
        <BrowserRouter>
            {component}
        </BrowserRouter>
    </StoreProvider>);
}

export * from '@testing-library/react';
export { customRender } ;