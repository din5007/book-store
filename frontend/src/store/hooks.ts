import { createTypedHooks } from "easy-peasy";

import { AppStoreModel } from ".";

export const { useStoreActions, useStoreState } = createTypedHooks<AppStoreModel>();