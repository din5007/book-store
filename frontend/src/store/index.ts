import { HttpService } from "../service/http/http.service";
import { createStore } from 'easy-peasy';
import { BookModel, booksModel } from "./books/books.model";
import { cartModel, CartModel } from "./cart/cart.model";
import { userModel, UserModel } from "./users/user.model";
import { OrderModel, orderModel } from "./order/order.model";

export interface AppStoreModel {
    books: BookModel;
    cart: CartModel;
    user: UserModel;
    order: OrderModel;
}

export const appStoreModel : AppStoreModel = {
    books: booksModel,
    cart: cartModel,
    user: userModel,
    order: orderModel,
}

export function storeModel() {
    return createStore(appStoreModel, {
        injections: {
            httpService: new HttpService()
        },
        initialState: {
            
        },
    })
}
export const store =  storeModel();