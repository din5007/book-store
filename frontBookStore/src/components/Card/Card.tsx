import { useEffect } from "react";
import { Row, Col } from "react-bootstrap";
import Button from "react-bootstrap/Button";
import Card from "react-bootstrap/Card";
import { useStoreActions, useStoreState } from "../../store/hooks";

function BookCard() {
  const { books } = useStoreState((state) => state.books);
  const { fetchAllBooks } = useStoreActions((actions) => actions.books);
  const { addToCart } = useStoreActions((actions) => actions.cart);
  
  useEffect(() => {fetchAllBooks({});}, []);

  return (
    <>
      <Row className="m-auto">
        {books?.map(({ id, title, author, price}) => (
          <Col key={id} className="m-auto py-3">
            <Card style={{ width : '18rem'}}>
              <Card.Body>
                <Card.Title>{title.toUpperCase()}</Card.Title>
                <Card.Text>{author.toUpperCase()}</Card.Text>
                <Card.Text>Price : {price} INR</Card.Text>
                <Button variant="primary" onClick={() => addToCart({ id })}>Add to Cart</Button>
              </Card.Body>
            </Card>
          </Col>
        ))}
      </Row>
    </>
  );
}

export default BookCard;
