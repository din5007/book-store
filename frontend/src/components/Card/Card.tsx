import { useEffect } from "react";
import { Row, Col } from "react-bootstrap";
import Button from "react-bootstrap/Button";
import Card from "react-bootstrap/Card";
import { useStoreActions, useStoreState } from "../../store/hooks";

function BookCard() {
  const { books } = useStoreState((state) => state.books);
  const { fetchAllBooks } = useStoreActions((actions) => actions.books);
  const { addToCart } = useStoreActions((actions) => actions.cart);
  
  useEffect(() => {fetchAllBooks();}, []);
  return (
    <>
      <Row xs={1} md={3} lg={4} className="g-4 m-auto">
        {books?.map(({ id, title, author, price, quantity}) => (
          <Col key={id}>
            <Card style={{ width: "18rem" }}>
              <Card.Body>
                <Card.Title>{title.toUpperCase()}</Card.Title>
                <Card.Text>{author.toUpperCase()}</Card.Text>
                <Card.Text>Price : {price} INR</Card.Text>
                <Button variant="primary" onClick={() => addToCart({ id })}>Add to Cart</Button>
              </Card.Body>
            </Card>
          </Col>
        ))}
      </Row>
    </>
  );
}

export default BookCard;
