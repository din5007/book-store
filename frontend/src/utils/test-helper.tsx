import { StoreProvider } from "easy-peasy";
import { render } from '@testing-library/react';
import { AppStoreModel, store } from "../store";
import { Router } from 'react-router-dom';
import { ReactNode } from "react";



// const provider : React.FC<{children : ReactNode, store: any}> = ({ children, store }) => {
//     return (
        
//     )
// }

function customRender(component : any , store: any) {

    return render(<StoreProvider store={store}>
        {component}
    </StoreProvider>);
}

export * from '@testing-library/react';
export { customRender } ;