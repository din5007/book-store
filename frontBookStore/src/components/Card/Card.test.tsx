import { createStore } from 'easy-peasy';
import { customRender as render, waitFor } from '../../utils/test-helper';
import Card from "./Card"
import { appStoreModel } from '../../store';
import { describe, expect, it, vi } from 'vitest';

const mock = {
    get: vi.fn()
};

const store = createStore(appStoreModel, {
    injections: {
        httpService : mock
    },
    initialState: {
        books : {
            books: [{ id : 1, title : 'Harry Potter', author: 'JK rowling', price: 400, quantity: 400}]
        }
    }
})

describe('Card', () => {
    it('Card should load properly - snapshot', () => {
        const {container}  = render(<Card/>, store);
        expect(container).toMatchSnapshot();
    });

    it('Card should contain proper fields', async () => {
        const { getByText }  = render(<Card/>, store);
        waitFor(() => {
            expect(getByText('HARRY POTTER')).toBeDefined();
            expect(getByText('JK ROWLING')).toBeDefined();
            expect(getByText('400')).toBeDefined();
            expect(getByText('Add to Cart')).toBeDefined();
        })
    });

    it('fetchAllBooks request should be triggered when component renders', async () => {
        render(<Card/>, store);
        expect(mock.get).toBeCalled();
    });
})