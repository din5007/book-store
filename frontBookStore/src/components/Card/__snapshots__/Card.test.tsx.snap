// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Card > Card should load properly - snapshot 1`] = `
<div>
  <div
    class="g-4 m-auto row row-cols-lg-4 row-cols-md-3 row-cols-1"
  >
    <div
      class="col"
    >
      <div
        class="card"
        style="width: 18rem;"
      >
        <div
          class="card-body"
        >
          <div
            class="card-title h5"
          >
            HARRY POTTER
          </div>
          <p
            class="card-text"
          >
            JK ROWLING
          </p>
          <p
            class="card-text"
          >
            Price : 
            400
             INR
          </p>
          <button
            class="btn btn-primary"
            type="button"
          >
            Add to Cart
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
